### PR TITLE
Add Makefile

### PR DESCRIPTION
**Pull request by yunkeying**
# Pull Request Template

## Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

**Test Configuration**:

- Firmware version:
- Hardware:
- Toolchain:
- SDK:

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have checked my code and corrected any misspellings

### CODE REVIEW

**Comment by helight** [comment]
把 pr 和 issue 关联起来，你这个 pr 是针对那个 issue 的，关联进来，只要把 issue 地址贴到 pr 中，不过最好是把 pr 地址贴到 issue 中，这样就可以关联起来

**Comment by helight** [comment]
1. 这个项目没有加入到流水线中，做自动化
2. 建议直接建立 mem_watcher 项目，page_fault作为其中一个功能

File: eBPF_Supermarket/Memory_Subsystem/mem_watcher/page_fault/page_fault.bpf.c [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+	u32 pid = bpf_get_current_pid_tgid();
+	u64 tm = bpf_ktime_get_ns();
+	u64 *tsp = bpf_map_lookup_elem(&time_map, &pid);
+	
+	if (tsp)
+		tm -= *tsp;
+	else
+		return 1;
+
+	unsigned key = tm / 10000000;
```
Comment (LinkinPF) [inline eBPF_Supermarket/Memory_Subsystem/mem_watcher/page_fault/page_fault.bpf.c]: magic number需要注释下或者用宏之类的处理一下比较好

File: eBPF_Supermarket/Memory_Subsystem/mem_watcher/page_fault/page_fault.c [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+{
+	/*
+	char file_name[200];
+
+	snprintf(file_name, sizeof(file_name), "%s_kern.o", argv[0]);
+	if (load_bpf_file(file_name)) {
+		printf("%s", bpf_log_buf);
+
+		return 1;
+	}
```
Comment (LinkinPF) [inline eBPF_Supermarket/Memory_Subsystem/mem_watcher/page_fault/page_fault.c]: 注释的代码删掉吧

**Review by helight — APPROVED** [review]
LGTM

这里 page_fault 还是一个独立的程序，并不是 mem_watcher程序的一部分，
下一步尽快建立mem_watcher程序

**Review by chenamy2017 — APPROVED** [review]

**Pull request merged**